newa/__init__.py

```diff
@@ -477,9 +479,93 @@ class Request(Cloneable, Serializable):
     id: str
     context: RecipeContext = field(factory=dict)
     environment: RecipeEnvironment = field(factory=dict)
+    compose: Optional[str] = None
     git_url: Optional[str] = None
     git_ref: Optional[str] = None
     tmt_path: Optional[str] = None
+    plan: Optional[str] = None
+
+    def fetch_details(self) -> None:
+        raise NotImplementedError
+
+    def generate_tf_exec_command(self) -> str:
+        cmdline_start = "NO_COLOR=yes testing-farm request --no-wait"
```

Comment:
I would highly recommend using a list and dictionary to construct the command line and environment variables, respectively. It's much easier to follow them in the code, for both linters and tracebacks, there is no need for placeholders like `context_params` - iterate over context, and if it's empty, nothing is added to the list. Not even the check whether `self.context` is empty is needed. And, especially, there is no need to even think about spaces used as separators - do we start each component with a space, or end with a space? Plus, `subprocess.run()` accepts command as a list too, and even has a parameter for environment variables.

```python
environment: dict[str, str] = {
    'NO_COLOR': 'yes'
}

command: list[str] = [
    'testing-farm', 'request', '--no-wait'
]

...

if self.git_ref:
    command += ['--git-ref', self.git_ref]

...

# process context
for k, v in self.context.items():
    command += ['-c', f'{k}="{v}"']

...

return command, environment


def initiate_tf_request(self) -> TFRequest:
    command, environment = self.generate_tf_exec_command()

    process = subprocess.run(
        ' '.join(command),
        env=environment,
        ...
```

Reply:
Makes very sense

Reply:
fixed